src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ResourceDictionary.cs

```diff
@@ -1779,6 +1827,35 @@ private object FetchResource(
             return GetValue(resourceKey, out canCache);
         }
 
+        private WeakReferenceList GetOrCreateWeakReferenceList(object resourceKey)
+        {
+            this._weakDeferredResourceReferencesMap ??= new();
+
+            if (this._weakDeferredResourceReferencesMap.TryGetValue(resourceKey, out var weakDeferredResourceReferences) is false)
```

Comment:
```suggestion
            if (!this._weakDeferredResourceReferencesMap.TryGetValue(resourceKey, out var weakDeferredResourceReferences))
```

This might solve it?

Reply:
@batzen is it okay to commit this?

Reply:
Will fix it that way as it seems to be the only way to fix it.
It's still a false positive and quite shocking to me that an IDE type warning get's such a basic thing wrong.
Could you escalate that internally at MS please?

Reply:
1. There are > 100 places in the WPF codebase using `== false` that don't trigger that warning. The rule talks about "non-constant Boolean expression" and the usages are mostly properties, but properties provide no guarantee to be constant.
2. Being forced to write `!` to check for `false` makes code less readable, in my opinion

Reply:
I'd check again. The only place is MilCodeGen runtime which is outside solution currently.

#10642

Reply:
I just don't care anymore. If the team prefers worse readability it's their decision.
Guess i just got frustrated because there is rarely a point in time when it's possible to even compile the WPF repo without "random" errors.
Can't even compile this PR without messages/errors like:
```
Native tools bootstrap failed
```

```
C:\Users\batze\.nuget\packages\microsoft.templateengine.authoring.tasks\10.0.100-preview.4.25206.2\build\Microsoft.TemplateEngine.Authoring.Tasks.targets(25,5): Template configuration message MV011: One or more postActions have a malformed or missing manualInstructions value. [C:\DEV\OSS_Own\d
otnet\wpf\packaging\Microsoft.Dotnet.Wpf.ProjectTemplates\Microsoft.DotNet.Wpf.ProjectTemplates.ArchNeutral.csproj]
```

```
perl : error : Setting locale failed. [C:\DEV\OSS_Own\dotnet\wpf\src\Microsoft.DotNet.Wpf\src\PresentationUI\PresentationUI.csproj]
perl : error : Please check that your locale settings: [C:\DEV\OSS_Own\dotnet\wpf\src\Microsoft.DotNet.Wpf\src\PresentationUI\PresentationUI.csproj]
        LC_ALL = (unset),
        (possibly more locale environment variables)
        LANG = "en_US.UTF-8"
      are supported and installed on your system.
perl : error : Falling back to the system default locale ("German_Germany.1252"). [C:\DEV\OSS_Own\dotnet\wpf\src\Microsoft.DotNet.Wpf\src\PresentationUI\PresentationUI.csproj]
```

Reply:
Guess nobody has bothered to run this on a german locale Windows. But I understand it is frustrating.

I use en-US and tbh its been over a year since I've encountered any issue besides one-time that was my fault.

Reply:
Yes, me too. I can not compile the WPF repo without "random" errors.

Reply:
>Could you escalate that internally at MS please?

Let me check this with the analyzer teams about this scenario and if it is in accordance to their design. Thank you for making this change as of now though.

>Guess i just got frustrated because there is rarely a point in time when it's possible to even compile the WPF repo without "random" errors.
Can't even compile this PR without messages/errors like:

We were not aware of such failure till now, would you please create a separate issue on this so that we can take a look at it? Will help in better tracking of this thread.